lib/include/event-driven/vDrawSkin.h

```diff
@@ -0,0 +1,1159 @@
+    {
+        std::list<unsigned int> res;
+        for (size_t i = 0; i < vec.size(); i++)
+        {
+            if (vec[i]==val)
+            {
+                res.push_back(i);
+            }
+        }
+        return res;
+    }
+
+    std::map<int, std::vector<double> > data;
+
+public:
+    int xoffset,yoffset;
+    int scaling =4, radius = 6;
+    int noise = 2500;
+    int max_value = 15000;//good for default flat skin patch
+
+    std::map<int, std::list<unsigned int> > repr2TaxelList;
+    virtual void initRepresentativeTaxels(std::vector<int> taxelMap){};
+    virtual int* getID(const int index) const {};
+    virtual std::tuple<double,double> makeMap( int ID, int taxelIntriangle) const {};
+    void get_data(int id,double x,double y, double orientation,double gain, int mirror,int layoutNum){
+        data.insert(std::make_pair(id,std::vector<double> {x,y,orientation,gain,double(mirror)}));
+    };
+};
+
+
+class Triangle_10pad : public PAD{
+
+private:
+
+
+int* getID(const int index)const {
+
+        static int result[3];
+        int triangleID,taxelInTriangle;
+        bool success =0;
+
+
+        for(auto mapit=repr2TaxelList.begin();mapit!=repr2TaxelList.end();mapit++){
+            auto listIntriangle= mapit->second;
+            auto it = std::find(listIntriangle.begin(), listIntriangle.end(), index);
+            if(it != listIntriangle.end()){
+                triangleID = mapit->first;
+                //taxelInTriangle = std :: distance(listIntriangle.begin(),it);
+                taxelInTriangle = index-16*(triangleID-3)/12;
+                success =1;
+                break;
+            }
+        }
+
+        if (triangleID==-1){
+            for(auto mapit=repr2TaxelList.begin();mapit!=repr2TaxelList.end();mapit++){
+            auto listIntriangle= mapit->second;
+            auto it = std::find(listIntriangle.begin(), listIntriangle.end(), index-1);
+            if(it != listIntriangle.end()){
+                triangleID = mapit->first;
+                break;
+                }
+            }
+            if(taxelInTriangle%2==0)taxelInTriangle=6;
+            else taxelInTriangle = 10;
+        }
+
+        triangleID = (triangleID-3)/12;
+        result[0] = triangleID;
+        result[1] = taxelInTriangle;
+        result[2] = success;
+        return result;
+    };
+
+
+ public :
+
+    Triangle_10pad(){
+
+        nTaxels=12;
+
+        xoffset = -200;
+        yoffset = 600;
+
+
+        dX = new double[nTaxels];
+        dY = new double[nTaxels];
+
+        dX[8]=-128.62; dY[8]=222.83;
+        dX[10]= 0; dY[10]=296.55;
+        dX[ 9]= 0; dY[ 9]=445.54;
+        dX[ 11]= 128.62; dY[ 11]=222.83;
+        dX[ 0]= 257.2; dY[ 0]=0;
+        dX[ 6]= -256.3; dY[ 6]=-147.64;
+        dX[ 1]=  385.83; dY[1]=-222.83;
+        dX[ 2]=128.62; dY[ 2]=-222.83;
+        dX[ 3]=0.0; dY[ 3]=0.0;
+        dX[ 4]=-128.62; dY[ 4]=-222.83;
+        dX[ 5]=-385.83; dY[ 5]=-222.83;
+        dX[ 7]=-257.2; dY[ 7]=0.0;
```

Comment:
Again a lot of hardcoded values and the constructor doesn't take any parameter. Maybe adding a file for managing the dx and dy could be an option. The others parameters should have an initialization via constructor. nTaxels is nothing more than the length of dX and dY. very error-prone initialization

Reply:
agreed, this should be fixed if/when loading from the file gets done. I added a deconstructor which deletes these correctly.